incubator/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/stream/MqttKafkaSubscribeFactory.java

```diff
@@ -569,13 +800,25 @@ private void doKafkaBegin(
             long authorization,
             long affinity)
         {
-            initialSeq = delegate.initialSeq;
-            initialAck = delegate.initialAck;
-            initialMax = delegate.initialMax;
-            state = MqttKafkaState.openingInitial(state);
+            if (!MqttKafkaState.initialOpening(state))
+            {
+                initialSeq = mqtt.initialSeq;
+                initialAck = mqtt.initialAck;
+                initialMax = mqtt.initialMax;
+                state = MqttKafkaState.openingInitial(state);
+                mqtt.currentFilters.forEach(filter ->
+                {
+                    int subscriptionId = (int) filter.subscriptionId();
+                    if (!mqtt.messagesSubscriptionIds.contains(subscriptionId))
+                    {
+                        mqtt.messagesSubscriptionIds.add(subscriptionId);
+                    }
+                });
```

Comment:
Try to avoid manipulating internal state of the other side if possible as it breaks the encapsulation.